File: Python/9. Palindrome Number.py
# 9. Palindrome Number
# Level: Easy
# https://leetcode.com/problems/palindrome-number/

class Solution:
    def isPalindrome(self, x: int) -> bool:
        if x < 0:
            return False
        
        xStr = str(x)
        pLeft = 0
        pRight = len(xStr) -1

        while pLeft < pRight:
            if xStr[pLeft] == xStr[pRight]:
                pLeft += 1
                pRight -= 1
                continue;
            return False
        return True
        

File: TypeScript/9. Palindrome Number.ts
// 9. Palindrome Number
// Level: Easy
// https://leetcode.com/problems/palindrome-number/

function isPalindrome(x: number): boolean {
    if (x < 0)
        return false;
    
    const xStr = x.toString();
    let pLeft = 0;
    let pRight = xStr.length - 1;

    while (pLeft < pRight){
        if(xStr[pLeft] === xStr[pRight]){
            pLeft++;
            pRight--;
            continue;
        }
        return false;
    }
    return true;
};